MapboxNavigation/NavigationViewController.swift

```diff
@@ -572,11 +584,45 @@ extension NavigationViewController: NavigationServiceDelegate {
         let advancesToNextLeg = componentsWantAdvance && (delegate?.navigationViewController(self, didArriveAt: waypoint) ?? defaultBehavior)
         
         if service.routeProgress.isFinalLeg && advancesToNextLeg && showsEndOfRouteFeedback {
-            showEndOfRouteFeedback()
+            // In case of final destination present end of route view first and then zoom in to extruded building.
+            showEndOfRouteFeedback { [weak self] _ in
+                self?.zoomInAndHighlightBuilding(for: service.router.location)
+            }
         }
         return advancesToNextLeg
     }
     
+    private func zoomInAndHighlightBuilding(for location: CLLocation?) {
+        if !highlightsDestinationBuildings { return }
+        guard let mapViewController = self.mapViewController else { return }
+        guard let location = location else { return }
+        
+        // Since all buildings are included in zoom level 16 and above as per
+        // https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/#building
+        // we make sure to zoom in before making update to insets, user course view etc
+        var currentZoomLevel = mapViewController.mapView.zoomLevel
+        let expectedZoomLevel = 16.5
+        if currentZoomLevel < expectedZoomLevel {
+            currentZoomLevel = expectedZoomLevel
+        }
+        
+        let mapView = mapViewController.mapView
+        mapView.setCenter(location.coordinate,
+                          zoomLevel: currentZoomLevel,
+                          direction: location.course,
+                          animated: true,
```

Comment:
Let’s verify that the camera gets reset correctly when beginning the next leg of the route.